### PR TITLE
Fix local storage race conditions

### LINKED ISSUE #378: [FE] Storage item race condition for auto-filled fields
Related to #377

Dropdown items like "Redeem", "Split Positions", "Merge Positions", etc. are links but at the same should auto complete some fields in the target form. This works well with regular left clicks (open link in the same window), but it doesn't work so well when using middle click to open multiple tabs.

**For example:**

- Go to any condition's details page.
- Open the top right dropdown.
- Always using middle click open "Split Position", "Merge Positions", and "Report Payouts".
- Check the opened tabs, and you'll see that only one (or maybe none) of the tabs has the pre-filled fields correctly pre-filled. 

This is because there's a unique identifier for the storage item used to transfer data, and that storage item is deleted when the section loads, making it unavailable for the other tabs.

We should fix this and all forms should be pre-filled as expected.

**Update 19/11/2020:**

@hpmaxi The latest changes to the condition and position selection are already in place. That means that part of this functionality (auto filling conditions / positions) could be broken. Please have that in mind when addressing this and fix everything you find. Also https://github.com/gnosis/conditional-tokens-explorer/issues/378#issuecomment-729901163

### PR DESCRIPTION
**Pull request by hpmaxi**
Closes #378 

This is using GET params, you can split from a condition but you can't from a position using the menu. This is much simpler and less error prone. 

~The solution consists of saving keys with the target operation prepended, which is not perfect but it works.~

~Merge was removed from the options because with the new design you can't select a condition before selecting a position. Local storage handling was removed from Context Providers of positions and conditions (Seems to be used only in details that don't need to read the local storage).~

~There is a missing piece but it doesn't look like it is related to local storage. There are some weird race conditions in `redeem`, if you put '7' and then another '7' to search for '77' sometimes it queries the graph first with '77' and then with '7', is that or there is something that filters that is not being used for toggling of the redeem option in the menu and that is all the problem.~

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr686--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr686--conditionaltokensexplorer.review.gnosisdev.com

**Review by gabitoesmiapodo — CHANGES_REQUESTED** [review]
It kinda works, but if I open several tabs using middle click some tabs don't get the relevant info.

Check below: I open four "Report Payouts" tabs. First two get the condition id, last two don't. If I wait for each tab to load instead of opening one after another, it seems to work, but not always.



The same happens for "Split Position" from the conditions' list.

For positions list:

- "Split": The same happens. Anyways, should we keep this option? Splitting starts from a condition.
- "Redeem": The same happens.

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr686--conditionaltokensexplorer.review.gnosisdev.com

File: src/components/table/SelectableConditionTable/index.tsx [hunk 1]
```diff
@@ -79,8 +80,10 @@ const SelectConditionTable: React.FC<Props> = (props) => {
     ...restProps
   } = props
 
-  const [textToSearch, setTextToSearch] = useState<string>('')
-  const [textToShow, setTextToShow] = useState<string>('')
+  const { conditionId } = useParams<Params>()
+
+  const [textToSearch, setTextToSearch] = useState<string>(conditionId || '')
```
Comment (hpmaxi) [inline src/components/table/SelectableConditionTable/index.tsx:85]: Initializes with parameter or empty string.

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr686--conditionaltokensexplorer.review.gnosisdev.com

**Review by gabitoesmiapodo — APPROVED** [review]
giphy

File: src/pages/PositionDetails/Contents.tsx [hunk 2]
```diff
@@ -341,20 +333,14 @@ export const Contents = (props: Props) => {
...
-        onClick: () => {
-          setValue(positionId)
-        },
-        text: 'Split',
-      },
+      // {
+      //   disabled: !userHasBalance || !isConnected,
+      //   href: `/split/${positionId}`,
+      //   text: 'Split',
+      // },
```
Comment (mariano-aguero) [inline src/pages/PositionDetails/Contents.tsx:343]: Why this is commented ?

File: src/pages/PositionsList/index.tsx [hunk 3]
```diff
@@ -222,21 +220,18 @@ export const PositionsList = () => {
...
-          text: 'Split',
         },
+        // {
+        //   disabled: !userHasERC1155Balance || !isConnected,
+        //   href: `/split`,
+        //   onClick: () => {
+        //     setValueSplit(id)
+        //   },
+        //   text: 'Split',
+        // },
```
Comment (mariano-aguero) [inline src/pages/PositionsList/index.tsx:234]: Same here

File: src/pages/ConditionsList/index.tsx [hunk 4]
```diff
@@ -226,34 +223,20 @@ export const ConditionsList: React.FC = () => {
...
-          },
           disabled: !isConnected,
         },
         {
-          href: `/merge/`,
-          text: 'Merge Positions',
-          onClick: () => {
-            setValue(id)
-          },
-          disabled: !isConnected,
```
Comment (mariano-aguero) [inline src/pages/ConditionsList/index.tsx:242]: Why this action was removed ?

**Review by mariano-aguero — COMMENTED** [review]
See comments

File: src/pages/PositionDetails/Contents.tsx (hunk 2 above)
Comment (hpmaxi) [reply src/pages/PositionDetails/Contents.tsx:343]: Parameters are passed by URL, so for the split, we should choose to interpret the id as a condition or as a position but not both. These lines could be deleted, though.
Also, we can use another format but that wasn't discussed when the thread about this issue came up.

File: src/pages/PositionsList/index.tsx (hunk 3 above)
Comment (hpmaxi) [reply src/pages/PositionsList/index.tsx:234]: Same answer

**Comment by gabitoesmiapodo** [comment]
@hpmaxi Delete the comments,  merge the PR.

File: src/pages/ConditionsList/index.tsx (hunk 4 above)
Comment (hpmaxi) [reply src/pages/ConditionsList/index.tsx:242]: You can't pre-select a condition with the new UI.

**Review by mariano-aguero — APPROVED** [review]

**Pull request merged**